AutoGear.lua

```diff
@@ -242,11 +256,22 @@ if TOC_VERSION_CURRENT < TOC_VERSION_MOP then
 		if (not numTalentTabs) or (numTalentTabs < 2) then
 			AutoGearPrint("AutoGear: numTalentTabs in AutoGearGetSpec() is "..tostring(numTalentTabs),0)
 		end
-		for i = 1, numTalentTabs do
-			local spec, _, pointsSpent = GetTalentTabInfo(i)
-			if (highestPointsSpent == nil or pointsSpent > highestPointsSpent) then
-				highestPointsSpent = pointsSpent
-				highestSpec = spec
+		-- isCata function where GetTalentTabInfo() index is different
+		if isCata then
+			for i = 1, numTalentTabs do
+				local _, spec, _, _, pointsSpent = GetTalentTabInfo(i)
+				if (highestPointsSpent == nil or pointsSpent > highestPointsSpent) then
+					highestPointsSpent = pointsSpent
+					highestSpec = spec
+				end
+			end
+		else
+			for i = 1, numTalentTabs do
+				local spec, _, pointsSpent = GetTalentTabInfo(i)
+				if (highestPointsSpent == nil or pointsSpent > highestPointsSpent) then
+					highestPointsSpent = pointsSpent
+					highestSpec = spec
+				end
 			end
 		end
```

Comment:
I would like to avoid the maintenance burden this implies.  If we check `isCata` here, behavior changes based on TOC instead of function interfaces.  In this case, I think we should do something more dynamic like checking the number of return values or the return type of `GetTalentTabInfo` instead.  That way, it won't stop working in the future, like a "Classic MoP", for example.

Maybe something like this:

```Lua
local _, spec, _, _, pointsSpent = GetTalentTabInfo(1)
if pointsSpent and pointsSpent >= 0 then
	highestPointsSpent = pointsSpent
	for i = 2, numTalentTabs do
		local _, spec, _, _, pointsSpent = GetTalentTabInfo(i)
		if (pointsSpent > highestPointsSpent) then
			highestPointsSpent = pointsSpent
			highestSpec = spec
		end
	end
else
	for i = 1, numTalentTabs do
		local spec, _, pointsSpent = GetTalentTabInfo(i)
		if (highestPointsSpent == nil or pointsSpent > highestPointsSpent) then
			highestPointsSpent = pointsSpent
			highestSpec = spec
		end
	end
end
```

With this, we'd try the 5-return call and if it's not nil and `pointsSpent` is a positive number, then it's pretty safe to say that it's using that 5-return-value implementation regardless of what expansion it is.  Otherwise, it'll use the 3-return-value implementation which still works elsewhere.

Reply:
implemented your code